File: app/src/main/kotlin/com/openclaw/ghostcrab/domain/model/ModelInfo.kt
package com.openclaw.ghostcrab.domain.model

import kotlinx.serialization.Serializable

/**
 * A language model known to the OpenClaw Gateway.
 *
 * @param id Unique identifier used by the gateway (e.g. `"gpt-4o"`).
 * @param provider Provider name (e.g. `"openai"`, `"anthropic"`).
 * @param displayName Human-readable label for UI.
 * @param isActive Whether this model is currently the active/default one.
 * @param status Operational status from the gateway (e.g. `"ready"`, `"auth-error"`).
 * @param capabilities Capability keys (e.g. `"vision"`, `"function-calling"`).
 */
@Serializable
data class ModelInfo(
    val id: String,
    val provider: String,
    val displayName: String,
    val isActive: Boolean,
    val status: String,
    val capabilities: List<String>,
)

File: ios/GhostCrab/Domain/Model/ModelInfo.swift
import Foundation

/// A language model known to the OpenClaw Gateway.
///
/// - Parameters:
///   - id: Unique identifier used by the gateway (e.g. `"gpt-4o"`).
///   - provider: Provider name (e.g. `"openai"`, `"anthropic"`).
///   - displayName: Human-readable label for UI.
///   - isActive: Whether this model is currently the active/default one.
///   - status: Operational status from the gateway (e.g. `"ready"`, `"auth-error"`).
///   - capabilities: Capability keys (e.g. `"vision"`, `"function-calling"`).
public struct ModelInfo: Codable, Sendable, Identifiable, Hashable {
    public let id: String
    public let provider: String
    public let displayName: String
    public let isActive: Bool
    public let status: String
    public let capabilities: [String]

    public init(
        id: String,
        provider: String,
        displayName: String,
        isActive: Bool,
        status: String,
        capabilities: [String]
    ) {
        self.id = id
        self.provider = provider
        self.displayName = displayName
        self.isActive = isActive
        self.status = status
        self.capabilities = capabilities
    }
}
